app/javascript/react/actions.es6

```diff
@@ -0,0 +1,191 @@
+import C from './constants'
+import Reception from '../routes.js.erb'
+
+export const matchSelection = (expected, observed) => {
+  return {
+    type: C.MATCH_SELECTION,
+    observed, expected
+  }
+}
+
+export const unmatch = (expected, observed) =>  {
+  return {
+    type: C.UNMATCH,
+    observed, expected
+  }
+}
+
+export const loadManifest = (manifest) => {
+  return {
+    type: C.LOAD_MANIFEST, manifest
+  }
+}
+
+export const selectExpectedOption = (val) => {
+  return {
+    type: C.SELECT_EXPECTED_OPTION, value: val
+  }
+}
+
+export const selectObservedOption = (val) => {
+  return {
+    type: C.SELECT_OBSERVED_OPTION, value: val
+  }
+}
+
+export const setManifestValue = (labwareId, address, fieldName, value, plateId) => {
+  return {
+    type: C.SET_MANIFEST_VALUE, labwareId, address, fieldName, value, plateId
+    //,
+    /*meta: {
+      debounce: {
+        time: 1000,
+        key: 'SET_MANIFEST_INPUT_DEBOUNCED'
+      }
+    }*/
+  }
+}
+
+export const filteredState = (state) => {
+  let dupState = Object.assign({}, state)
+
+  if (Object.keys(dupState.mapping).length == 0) {
+    delete dupState.mapping
+  }
+}
+
+export const cacheTaxonomy = (taxId, data) => {
+  let obj = {}
+  obj[taxId] = data
+
+  return {
+    type: C.CACHE_TAXONOMY,
+    data: obj
+  }
+}
+
+export const updateScientificName = (labwareId, address, fieldName, taxId, plateId, taxonomyServiceUrl) => {
+  return (dispatch, getState) => {
+    if (taxId.length == 0) {
+      dispatch(setManifestValue(labwareId, address, fieldName, "", plateId))
+      return
+    }
+    const cache = getState().services.cachedTaxonomies
+    if (cache) {
+      const val = cache[taxId]
+      if (val) {
+        dispatch(setManifestValue(labwareId, address, fieldName, val.scientificName, plateId))
+        return
+      }
+    }
+    return $.ajax(taxonomyServiceUrl+'/'+taxId, {
```

Comment:
Found where you use the service URL!

Reply:
Thanks, for some reason I thought this was different. Then it is still needed